### PR TITLE
Routing system

### PR DESCRIPTION
**Pull request by amalexandrescu**
Creates an alert if the current page location doesn't match with the current user role.

### CODE REVIEW

**Review by BowTiedDeployer — DISMISSED** [review]

File: src/components/ConnectWallet.tsx [hunk 1]
```diff
@@ -18,15 +18,34 @@ const ConnectWallet = ({ currentTheme }: ConnectWalletProps) => {
...
+  useEffect(() => {
+    controlAccessRoutes();
+  }, [location]);
+
   const disconnect = () => {
     dispatch(disconnectAction());
   };
 
-  const authenticate = () => {
+  const authenticate = async () => {
```
Comment (Alexandrescu) [inline src/components/ConnectWallet.tsx]: No need to async here

File: src/components/ConnectWallet.tsx [hunk 2]
```diff
@@ -18,15 +18,34 @@ const ConnectWallet = ({ currentTheme }: ConnectWalletProps) => {
...
     dispatch(connectAction());
   };
 
   if (userSession.isUserSignedIn()) {
+    if (currentRole === 'Viewer') {
```
Comment (Alexandrescu) [inline src/components/ConnectWallet.tsx:45]: Please add another role "None" which is going to be the initial state and add it here.

File: src/components/ConnectWallet.tsx [hunk 3]
```diff
@@ -18,15 +18,34 @@ const ConnectWallet = ({ currentTheme }: ConnectWalletProps) => {
...
+      dispatch(updateUserRoleAction());
+      return <div>Loading role...</div>;
```
Comment (Alexandrescu) [inline src/components/ConnectWallet.tsx:47]: Make sure this is looking decent.

File: src/components/LeftPanel.tsx [hunk 4]
```diff
@@ -67,39 +86,107 @@ const LeftPanel = ({ currentTheme }: ConnectWalletProps) => {
...
-            <ListItemButton component={Link} to={'/cinema'}>
-              <ListItemIcon>
-                <Movie style={{ color: colors[currentTheme].secondary }} />
-              </ListItemIcon>
-              <ListItemText style={{ color: colors[currentTheme].secondary }} primary="Cinema" />
+              <ListItemText style={{ color: colors[currentTheme].secondary }} primary="Dashboard" />
             </ListItemButton>
           </ListItem>
         </div>
+        {currentRole !== 'Viewer' && (
```
Comment (Alexandrescu) [inline src/components/LeftPanel.tsx:104]: Was this `===`? Or is it correct? What happens for the other roles?

File: src/components/appMenuSections/profile/Profile.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import MinerProfile from './MinerProfile';
+import PendingMinerProfile from './PendingMinerProfile';
+import WaitingMinerProfile from './WaitingMinerProfile';
+
+const Profile = () => {
+  const currentRole: UserRole = useAppSelector(selectCurrentUserRole);
+  return (
+    <div>
+      <h2>Profile</h2>
+      {currentRole !== 'Miner' && <CommonInfoProfile />}
```
Comment (Alexandrescu) [inline src/components/appMenuSections/profile/Profile.tsx]: Pro tip: You can create a
```
const profileComponent = Record<UserRole, React.Component> = {
   'Viewer': <CommonInfoProfile>,
   'Miner': <MinerProfile>
    ...
}
```

and use this here ```profileComponent[currentRole]``` in the html

File: src/redux/actions/index.ts [hunk 6]
```diff
@@ -8,3 +13,23 @@ export const disconnectAction = () => {
 export const connectAction = () => {
   return { type: CONNECT_USER_SESSION };
 };
+
+// export const updateUserRoleAction = (newRole: string) => {
```
Comment (Alexandrescu) [inline src/redux/actions/index.ts]: Cleanup

File: src/redux/actions/index.ts [hunk 7]
```diff
@@ -8,3 +13,23 @@ export const disconnectAction = () => {
...
+
+const updateRole = (): Promise<UserRole> => {
+  return Promise.resolve('NormalUser');
+};
+
+export const updateUserRoleAction = () => {
+  return async (dispatch: AppDispatch, getState: () => IinitialState) => {
+    try {
+      const userRole = await updateRole();
+      // dispatch(updateUserRoleAction(userRole));
```
Comment (Alexandrescu) [inline src/redux/actions/index.ts]: Cleanup

**Review by Alexandrescu — DISMISSED** [review]

File: src/components/LeftPanel.tsx (hunk 4 above)
Comment (amalexandrescu) [reply src/components/LeftPanel.tsx:104]: The default role is Viewer. You are supposed to have access only to the dashboard if you are not logged in. Access to the profile you have when you are from NormalUser above.

File: src/components/appMenuSections/profile/Profile.tsx [hunk 8]
```diff
@@ -7,15 +7,16 @@ import WaitingMinerProfile from './WaitingMinerProfile';
 
 const Profile = () => {
   const currentRole: UserRole = useAppSelector(selectCurrentUserRole);
-  return (
-    <div>
-      <h2>Profile</h2>
```
Comment (BowTiedDeployer) [inline src/components/appMenuSections/profile/Profile.tsx]: missing after conversion to component

**Review by BowTiedDeployer — APPROVED** [review]

**Review by BowTiedDeployer — APPROVED** [review]

**Pull request merged**